### PR TITLE
Use dse_version in protocol negotiation

### PR DESCRIPTION
**Pull request by tolbertam**
* Parses dse_version from system.local and peers table and if present
  stores as Host.dseVersion.
* If > DSE 5.1 is detected, consider DSE_Vx protocol versions in
  finding highest common protocol version.

Needs more testing (long test and simulacron tests), but this should work good I hope.

### CODE REVIEW

**Comment by tolbertam** [comment]
Adjusted long test to test downgrade from DSE 5.1.5 to 5.0.11.  One observation from that is that it appears DSE 5.1 was the first dse release to include `dse_version` in the peers/local table, so before that version the regular cassandra comparison will kick in (which is fine).

Unfortunately I don't currently have a way to test DSE protocol versions with simulacron (will eventually), but I don't think it's needed for now as the unit tests validate the downgrade scenarios pretty adequately.

Removing wip tag as I think this is ready.

**Comment by jorgebay** [comment]
> I don't think it's needed for now as the unit tests validate the downgrade scenarios pretty adequately

I agree.

File: test/integration/long/control-connection-tests.js [hunk 1]
```diff
@@ -14,20 +14,20 @@ const Client = require ('../../../lib/client');
 describe('Control Connection', function () {
   this.timeout(240000);
   describe('#init()', function () {
-    after(helper.ccmHelper.removeIfAny);
-    it('should downgrade to protocol 2 with versions 2.1 & 2.0', function (done) {
+    //after(helper.ccmHelper.removeIfAny);
```
Comment (jorgebay) [inline test/integration/long/control-connection-tests.js]: Probably unintended?

Comment (tolbertam) [reply test/integration/long/control-connection-tests.js]: woops, yep!  I'll fix that.

**Comment by jorgebay** [comment]
I'll fix the commented out line and merge it so we can use the new repo.

**Pull request merged**